### PR TITLE
Move regression tests to a new format

### PR DESCRIPTION
**Pull request by reverendbedford**
Blackbear/MOOSE would like csv format tests they can use to make sure that running NEML through MOOSE produces the same results as directly running the model in NEML.  They could use our regression tests to do this, provided we make some changes to the format.  This PR moves over the regression tests to this new format, now in test/test_regression.

### CODE REVIEW

File: test/test_regression/regressiontools.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  data = np.concatenate((np.array(driver.t_int)[:,None], 
+    np.array(driver.T_int)[:,None],
+    np.array(driver.mechanical_strain_int),
+    np.array(driver.stress_int)), axis = 1)
+
+  return data
+
+def rtt(tdir):
+  model = parse.parse_xml(os.path.join(tdir, default_xml_file),
```
Comment (arovinelli) [inline test/test_regression/regressiontools.py:48]: can we add a description for this method

File: test/test_regression/test_regression.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import unittest
+
+sys.path.append(os.path.abspath(os.path.dirname(__file__)))
+from regressiontools import rtt
+
+def test_all_regression():
+  tests = glob.glob(os.path.join(os.path.dirname(__file__),"test_*"))
+  npe = glob.glob(os.path.join(os.path.dirname(__file__),"test*.py"))
+  for n in npe:
+    tests.remove(n)
```
Comment (arovinelli) [inline test/test_regression/test_regression.py:19]: not really sure if you are removing everything or just something from the tests, please add a comment explaining why remove is necessary

**Review by arovinelli — CHANGES_REQUESTED** [review]
just two minor comments

File: test/test_regression/regressiontools.py (hunk 1 above)
Comment (reverendbedford) [reply test/test_regression/regressiontools.py:52]: Yep, it actually runs the test (added description)

File: test/test_regression/test_regression.py (hunk 2 above)
Comment (reverendbedford) [reply test/test_regression/test_regression.py:21]: It removes the test file itself (has to be named test_*.py) from the list of directories with tests.

**Review by arovinelli — APPROVED** [review]
Please wait for the tests to finish before merging

**Pull request merged**